src/compiler/core.ts

```diff
@@ -990,10 +1020,10 @@ namespace ts {
         }
 
         return "^(" + pattern + (usage === "exclude" ? ")($|/)" : ")$");
-    }
 
-    function replaceWildcardCharacter(match: string) {
-        return match === "*" ? "[^/]*" : match === "?" ? "[^/]" : "\\" + match;
+        function replaceWildcardCharacter(match: string) {

```

Comment:
I'd prefer this stay outside of `getRegularExpressionForWildcard` and instead pass `singleAsteriskRegexFragment` as an argument to avoid the closure.